example/lib/devices_list/devices_bloc.dart

```diff
@@ -86,9 +86,30 @@ class DevicesBloc {
       if (_locationPermissionStatus != PermissionStatus.granted) {
         return Future.error(Exception("Location permission not granted"));
       }
+    } else if (Platform.isIOS) {
+      /// On iOS the user will be asked to allow Bluetooth access by
+      /// the system. Once it is granted, the Bluetooth state will change from
+      /// UNKNOWN to POWERED_ON. The following function will block the thread
+      /// until it happens.
+      await _waitForBluetoothPoweredOn();
     }
   }
 
+  Future<void> _waitForBluetoothPoweredOn() async {
```

Comment:
Shouldn't it be in Android related code as well? The scan would not start if bluetooth is off

Reply:
Implementing this I was having in mind only handling Bluetooth permission on iOS. But you're right it can be used on Android as well, for case when Bluetooth is really turned off and start scanning only after switching it on. I just tested it and it works just like described.